server/src/main/java/io/spine/server/integration/ThirdPartyContext.java

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright 2019, TeamDev. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.server.integration;
+
+import com.google.protobuf.Any;
+import io.spine.base.EventMessage;
+import io.spine.core.ActorContext;
+import io.spine.core.Event;
+import io.spine.core.EventContext;
+import io.spine.core.EventId;
+import io.spine.core.UserId;
+import io.spine.server.BoundedContext;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+import static io.spine.base.Identifier.newUuid;
+import static io.spine.base.Time.currentTime;
+import static io.spine.protobuf.AnyPacker.pack;
+import static io.spine.server.BoundedContextBuilder.notStoringEvents;
+
+/**
+ * An external upstream system which is not implemented in Spine.
```

Comment:
Maybe it's better to use `non-Spine-based` or `not based on Spine` if I got the meaning correctly.